hcipy/optics/segmented_mirror.py

```diff
@@ -0,0 +1,237 @@
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+
+from .optical_element import OpticalElement
+from ..aperture import circular_aperture, hexagonal_aperture, make_segmented_aperture
+from ..field import Field, make_hexagonal_grid, make_pupil_grid, evaluate_supersampled
+from ..io import write_fits
+from ..plotting import imshow_field
+
+
+pupil_size = 1024
+PUP_DIAMETER = 15.   # m
+
+
+def get_atlast_aperture(normalized=False, with_segment_gaps=True, segment_transmissions=1, write_to_disk=False, outDir=None):
+    """Make the ATLAST/HiCAT pupil mask.
+
+    This function is a copy of make_hicat_aperture(), except that it also returns the segment positions.
+
+    Parameters
+    ----------
+    normalized : boolean
+        If this is True, the outer diameter will be scaled to 1. Otherwise, the
+        diameter of the pupil will be 15.0 meters.
+    with_segment_gaps : boolean
+        Include the gaps between individual segments in the aperture.
+    segment_transmissions : scalar or array_like
+        The transmission for each of the segments. If this is a scalar, this transmission will
+        be used for all segments.
+
+    Returns
+    -------
+    Field generator
+        The ATLAST aperture.
+    CartesianGrid
+        The segment positions.
+    """
+    pupil_diameter = PUP_DIAMETER
+    segment_circum_diameter = 2 / np.sqrt(3) * pupil_diameter / 7
+    num_rings = 3
+    segment_gap = 0.01   # m
+
+    if not with_segment_gaps:
+        segment_gap = 0
+
+    if normalized:
+        segment_circum_diameter /= pupil_diameter
+        segment_gap /= pupil_diameter
+        pupil_diameter = 1.0
+
+    segment_positions = make_hexagonal_grid(segment_circum_diameter / 2 * np.sqrt(3), num_rings)
+    segment_positions = segment_positions.subset(lambda grid: ~(circular_aperture(segment_circum_diameter)(grid) > 0))
+
+    hexagon = hexagonal_aperture(segment_circum_diameter - segment_gap)
+
+    def segment(grid):
+        return hexagon(grid.rotated(np.pi/2))
+
+    segmented_aperture = make_segmented_aperture(segment, segment_positions, segment_transmissions)
+
+    def func(grid):
+        res = segmented_aperture(grid)
+
+        return Field(res, grid)
+
+    # Save pupil to disk, as pdf and fits
```

Comment:
The saving to disk can probably be taken out, I just realized I left it in there because I needed it for my code.

Reply:
Yes, take this out.